src/app/clusters/dishwasher-alarm-server/dishwasher-alarm-server.cpp

```diff
@@ -0,0 +1,397 @@
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: writing  latch, err:0x%x", status);
+        return status;
+    }
+
+    return status;
+}
+
+EmberAfStatus DishwasherAlarmServer::SetStateValue(EndpointId endpoint, const BitMask<AlarmMap> newState, bool ignoreLatchState)
+{
+    BitMask<AlarmMap> supported;
+    BitMask<AlarmMap> finalNewState;
+    finalNewState.Set(newState);
+
+    if (GetSupportedValue(endpoint, &supported) || !supported.HasAll(finalNewState))
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: Alarm is not supported");
+        return EMBER_ZCL_STATUS_FAILURE;
+    }
+
+    BitMask<AlarmMap> mask;
+    if (GetMaskValue(endpoint, &mask) || !mask.HasAll(finalNewState))
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: Alarm is suppressed");
+        return EMBER_ZCL_STATUS_FAILURE;
+    }
+
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    BitMask<AlarmMap> currentState;
+    status = Attributes::State::Get(endpoint, &currentState);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: reading  state, err:0x%x", status);
+        return status;
+    }
+
+    BitMask<AlarmMap> latch;
+    if (!ignoreLatchState && (GetLatchValue(endpoint, &latch) == EMBER_ZCL_STATUS_SUCCESS))
+    {
+        // only reset bits which can be reset based on latch value
+        BitMask<AlarmMap> finalState(latch & currentState, finalNewState);
+        finalNewState = finalState;
+    }
+
+    status = Attributes::State::Set(endpoint, finalNewState);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: writing  state, err:0x%x", status);
+        return status;
+    }
+
+    // Generate Notify event.
+    BitMask<AlarmMap> becameActive;
+    becameActive.Set(finalNewState).Clear(currentState);
+    BitMask<AlarmMap> becameInactive;
+    becameInactive.Set(currentState).Clear(finalNewState);
+
+    SendNotifyEvent(endpoint, becameActive, becameInactive, finalNewState, mask);
+    return status;
+}
+
+EmberAfStatus DishwasherAlarmServer::ResetStateValue(EndpointId endpoint, const BitMask<AlarmMap> alarms)
+{
+    BitMask<AlarmMap> supported;
+    if (GetSupportedValue(endpoint, &supported) || !supported.HasAll(alarms))
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: Alarm is not supported");
+        return EMBER_ZCL_STATUS_FAILURE;
+    }
+
+    BitMask<AlarmMap> state;
+    if (GetStateValue(endpoint, &state) != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return EMBER_ZCL_STATUS_FAILURE;
+    }
+
+    state.Clear(alarms);
+    return SetStateValue(endpoint, state, true);
+}
+
+bool DishwasherAlarmServer::HasResetFeature(EndpointId endpoint)
+{
+    uint32_t featureMap = 0;
+    if (Attributes::FeatureMap::Get(endpoint, &featureMap) != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return false;
+    }
+
+    if (featureMap == 1)
```

Comment:
It should be `to_underlying(Feature::kReset)` except we don't have that in our XML for some reason that escapes me.  Please do this as a followup?

Reply:
Issue created for followup.